docs/content/parameters.md

```diff
@@ -0,0 +1,59 @@
+---
+title: Parameters
+description: The lifecycle of a parameter from definition, to resolution, and finally injection at runtime
+aliases:
+- /how-parameters-work/
+---
+
+When you are authoring a bundle, you can define what parameters your bundle
+requires such as username and password values for a backing database, or the
+region that a certain resource should be deployed in, etc. Then in your
+action's steps you can reference the parameters using porter's template
+language `{{ bundle.parameters.db_name }}`.
+
+Parameter values are resolved from a combination of supplied parameter set
+files, user-specified overrides and defaults defined by the bundle itself.
+The resolved values are added to a claim receipt, which is passed in to
+the bundle execution environment, e.g. the docker container, when the bundle
+action is executed (install/upgrade/uninstall/invoke).
+
+## Parameter Sets
+
+A Parameter Set is a file which maps parameters to their strategies for value
+resolution.  Strategies include resolving from a source on the user's machine
+such as an environment variable (`env`), filepath (`path`), command result
+(`command`) or simply the value itself (`value`).  In addition, a parameter
+can have a secret source (`secret`).  See the [secrets
+plugin docs](/plugins/types/#secrets) to learn how to configure Porter to use
+an external secret store.
+
+TODO: add docs on generate when functionality is added.
```

Comment:
Let's make this a html comment or just remove it so that we can publish it immediately without the TODO showing up on the final doc page.